src/charm.py

```diff
@@ -64,7 +71,7 @@ def __init__(self, *args) -> None:
         self.framework.observe(self.on.nssf_pebble_ready, self._configure_nssf)
         self.framework.observe(self.on.fiveg_nrf_relation_joined, self._configure_nssf)
         self.framework.observe(self._nrf_requires.on.nrf_available, self._configure_nssf)
-        self.framework.observe(self._nrf_requires.on.nrf_broken, self._on_nrf_broken)
+        self.framework.observe(self._nrf_requires.on.nrf_broken, self._configure_nssf)
```

Comment:
is it worth it to send this event to _configure_nssf, since it would just return due to the missing nrf data ?

Reply:
It provides to set the charm status as Blocked as soon as possible. 
When the nrf relation broken,  _configure_nssf is called. This runs self.ready_to_configure() which returns False because of `fiveg-nrf` relation does not exist. (By the way, broken relation could be detected within latest ops library changes). Then  _configure_nssf method exits. After every hook, _on_collect_unit_status runs repeatedly and  that sets the charm status to  Blocked(Waiting for fiveg-nrf relation).

Reply:
This is tested by removing observation of nrf_relation_broken event and def _on_collect_unit_status is called after every hook (no matter it is observed or not). So this line is not necessary and safe to remove.

Reply:
Done